### DISCUSSION TITLE
Implement proper ReconstructSome for leopard

### DISCUSSION THREAD

**Discussion by liamsi** [klauspost/reedsolomon#263] [Ideas/Features]
Currently, `ReconstructSome` for leopard just calls `ReconstructData`: 
https://github.com/klauspost/reedsolomon/blob/674f18bb1f8189237afb301da140a899a4a85756/leopard.go#L335-L337

For other implementations, the user can actually specify the required shards to be reconstructed.

I'm advocating to implement the same behaviour for leopard as well.

**Comment by klauspost** [comment]
It surprises me more that it doesn't reconstruct all shards, but only data. 

It seems like the bitfield stuff would make it a feasible change, though due to the progressive encoding extra work would usually have to be done.

**Comment by odeke-em** [comment]
I am working on this issue @liamsi @klauspost
